t/plugin/ip-restriction.t

```diff
@@ -0,0 +1,276 @@
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say(require("cjson").encode(conf))
+        }
+    }
+--- request
+GET /t
+--- response_body_like eval
+qr@invalid cidr range: Invalid prefix: /38@
+--- no_error_log
+[error]
+
+
+=== TEST 4: empty conf
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.ip-restriction")
+
+            local ok, err = plugin.check_schema({})
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+invalid "oneOf" in docuement at pointer "#"
+done
+--- no_error_log
+[error]
+
+
+=== TEST 5: empty CIDRs
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.ip-restriction")
+
+            local ok, err = plugin.check_schema({blacklist={}})
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+invalid "type" in docuement at pointer "#/blacklist"
+done
+--- no_error_log
+[error]
+
+
+=== TEST 6: whitelist and blacklist mutual exclusive
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.ip-restriction")
+            local ok, err = plugin.check_schema({whitelist={"172.17.40.0/24"}, blacklist={"10.255.0.0/16"}})
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+invalid "oneOf" in docuement at pointer "#"
+done
+--- no_error_log
+[error]
+
+
+=== TEST 7: set route(id: 1)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "uri": "/server_port",
+                        "upstream": {
+                            "type": "roundrobin",
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            }
+                        },
+                        "plugins": {
+                            "ip-restriction": {
+                                 "whitelist": [
+                                     "127.0.0.0/24"
+                                 ]
+                            }
+                        }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- yaml_config eval
+$::yaml_config
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: hit route
+--- request
+GET /server_port
+--- response_body_like eval
+qr/1980/
+--- no_error_log
+[error]
+
+
```

Comment:
should add test case for ip not in the whitelist.

Reply:
You're right, but the `nginx.conf` for test which is from `t/APISix.pm` doesn't have `real_ip_header` directive, do you have any idea or just add this in `t/APISix.pm`？

Reply:
you can mock the ip like this:
```
--- http_config
set_real_ip_from 127.0.0.1;
real_ip_header X-Forwarded-For;
--- more_headers
X-Forwarded-For: 113.74.26.106
```
the ip will be `113.74.26.106` now.

Reply:
Got it!